engine/src/cmd/damageable.h

```diff
@@ -26,78 +26,142 @@
 #ifndef DAMAGEABLE_H
 #define DAMAGEABLE_H
 
-#include "unit_armorshield.h"
+#include "damageable_object.h"
 #include "gfx/vec.h"
 
+class Unit;
+class GFXColor;
 
-class Damageable
+/**
+ * @brief The Damageable class TODO
+ */
+class Damageable : public DamageableObject
 {
 public:
-    // TODO: Consider making into protected later
-    // Fields
-    // Inconsistency between this (0) and Init (1) for hull and maxhull.
-    // Chose Init and value of 1. Also commented out was 10.
-    // Doesn't really matter, but documenting anyway.
-    float maxhull = 1;
-    float hull = 1;
-    Armor armor;
-    Shield shield;
+    DamageableLayer *hull;
+    DamageableLayer *armor;
+    DamageableLayer *shield;
+
+    float *current_hull;
+    float *max_hull;
+
 
+    bool dying;
+    //Is dead already?
+    bool  killed;
```

Comment:
would an enum be better than 2 booleans?
```
enum LifeStatus { ALIVE, DYING, DEAD };
state LifeStatus;
```

Reply:
The problem is the underlying "death" code is unclear and complicated. You would think a unit would just die and be gone. Instead, there's a dying process when it's exploding. This leads to all sorts of complicated code that I'll admit I don't fully understand. 
It would probably be better to kill the Unit object and have it eject a DyingUnit object along with cargo and pilot.
It also seems I never used dying, so I removed it.

Reply:
I agree with @royfalk here. A unit dying is quite the process. I don't fully understand it either

Reply:
okay.

I'm guessing the "death" code has to do with explosions and such. But we can leave it for now and come back to it in a later refactor.